plugins/destination/postgresql/client/insert.go

```diff
@@ -87,21 +90,41 @@ func (c *Client) InsertBatch(ctx context.Context, messages message.WriteInserts)
 }
 
 func (c *Client) flushBatch(ctx context.Context, batch *pgx.Batch) error {
-	if batch.Len() == 0 {
-		return nil
-	}
-	err := c.conn.SendBatch(ctx, batch).Close()
-	if err == nil {
+	var attempt int64
+
+	err := retry.Do(func() error {
+		if batch.Len() == 0 {
+			return nil
+		}
+		err := c.conn.SendBatch(ctx, batch).Close()
+		if err != nil {
+			return err
+		}
+
 		return nil
-	}
+	}, retry.RetryIf(func(err error) bool {
+		var pgErr *pgconn.PgError
+		if errors.As(err, &pgErr) {
+			if pgErr.Code == "40P01" && attempt < c.spec.RetryOnDeadlock {
```

Comment:
Don't think we need to track attempts and add jitter, the library has a default configuration https://github.com/avast/retry-go/blob/cbf4dbfb37a5861d7304d50e5e87b6ae72569dde/retry.go#L225

We can override it if needed

Reply:
Simplified in d21258e1e2.

Reply:
`lastErrorOnly` in a1d68b4d36